pkg/operator/ceph/cluster/mon/drain_test.go

```diff
@@ -0,0 +1,120 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mon
+
+import (
+	"context"
+	"testing"
+
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"github.com/rook/rook/pkg/operator/test"
+	"github.com/stretchr/testify/assert"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	mockPDBNamespace = "test-ns"
+)
+
+func createMonPDB(t *testing.T, ctx *clusterd.Context) error {
+	pdbObj := &policyv1beta1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      monPDBName,
+			Namespace: mockPDBNamespace,
+		},
+		Spec: policyv1beta1.PodDisruptionBudgetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{k8sutil.AppAttr: AppName},
+			},
+			MaxUnavailable: &intstr.IntOrString{IntVal: 1},
+		},
+	}
+
+	_, err := ctx.Clientset.PolicyV1beta1().PodDisruptionBudgets(mockPDBNamespace).Create(context.TODO(), pdbObj, metav1.CreateOptions{})
+
+	return err
+}
+func TestAllowMonDrain(t *testing.T) {
+	clientset := test.New(t, 1)
+	ctx := &clusterd.Context{
+		Clientset: clientset,
+	}
+	tests := []struct {
+		name         string
+		createMonPDB bool
+		wantErr      bool
+	}{
+		{"failed to allow mon drain", false, true},
+		{"allow mon drain", true, false},
+	}
+
+	for _, tt := range tests {
+		if tt.createMonPDB {
+			err := createMonPDB(t, ctx)
+			assert.NoError(t, err)
+		}
+		t.Run(tt.name, func(t *testing.T) {
+			// change MaxUnavailable mon PDB to 1
+			err := allowMonDrain(ctx, mockPDBNamespace)
+			if tt.wantErr {
+				assert.Error(t, err)
+				return
+			}
+			assert.NoError(t, err)
+			existingPDB, err := ctx.Clientset.PolicyV1beta1().PodDisruptionBudgets(mockPDBNamespace).Get(context.TODO(), monPDBName, metav1.GetOptions{})
+			assert.NoError(t, err)
+			assert.Equal(t, 1, int(existingPDB.Spec.MaxUnavailable.IntValue()))
+		})
+	}
+
+}
+
+func TestBlockMonDrain(t *testing.T) {
+	clientset := test.New(t, 1)
+	ctx := &clusterd.Context{
+		Clientset: clientset,
+	}
+	tests := []struct {
+		name         string
+		createMonPDB bool
+		wantErr      bool
+	}{
+		{"failed to block mon drain", false, true},
+		{"block mon drain", true, false},
+	}
+	for _, tt := range tests {
+		if tt.createMonPDB {
+			err := createMonPDB(t, ctx)
+			assert.NoError(t, err)
+		}
+		t.Run(tt.name, func(t *testing.T) {
+			// change MaxUnavailable mon PDB to 0
+			err := blockMonDrain(ctx, mockPDBNamespace)
+			if tt.wantErr {
+				assert.Error(t, err)
+				return
+			}
+			assert.NoError(t, err)
+			existingPDB, err := ctx.Clientset.PolicyV1beta1().PodDisruptionBudgets(mockPDBNamespace).Get(context.TODO(), monPDBName, metav1.GetOptions{})
+			assert.NoError(t, err)
+			assert.Equal(t, 0, int(existingPDB.Spec.MaxUnavailable.IntValue()))
+		})
+	}
```

Comment:
I think this is a lot more confusing to read with the loop. Why not create two separate calls to `t.Run` instead and do the setup for each as needed. I posted another comment with a link where you can see what I mean.

Reply:
^ https://github.com/rook/rook/pull/7442#discussion_r601608664

Reply:
Thanks for sharing the link. I've updated the tests.